### PR TITLE
Viz - Skip widgets redrawing when size changed on small value(T1156890)

### PR DESCRIPTION
**Pull request by alexanderPolosatov**

### CODE REVIEW

File: js/viz/core/base_widget.js [hunk 1]
```diff
@@ -112,6 +113,14 @@ function callForEach(functions) {
     functions.forEach(c => c());
 }
 
+function floorDimensionsValuesInCanvas(canvas) {
```
Comment (ksercs) [inline js/viz/core/base_widget.js]: Isn't it better to name it like this?
```suggestion
function floorCanvasDimensions(canvas) {
```

File: testing/tests/DevExpress.viz.charts/chartInteraction.tests.js [hunk 2]
```diff
@@ -730,3 +730,35 @@ QUnit.test('two stub axis', function(assert) {
...
+    beforeEach() {
+        this.$container = $('#chart');
+    },
+    createChart(options) {
+        return this.$container.dxChart(options).dxChart('instance');
+    }
+}, () => {
+    [-1, 1].forEach(sign => {
+        ['height', 'width'].forEach(dimension => {
+            QUnit.test(`Widget should not re-rendered when ${dimension} ${sign > 0 ? 'increased' : 'decreased'} on value less threshold`, function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.charts/chartInteraction.tests.js]:
```suggestion
            QUnit.test(`chart should not re-render when ${dimension} ${sign > 0 ? 'increased' : 'decreased'} on value less threshold`, function(assert) {
```

File: testing/tests/DevExpress.viz.charts/chartInteraction.tests.js [hunk 3]
```diff
@@ -730,3 +730,35 @@ QUnit.test('two stub axis', function(assert) {
...
+        
+                drawnHandler.reset();
+            
+                chart.option(`size.${dimension}`, initialSize[dimension] + sign * 0.098);
+            
+                assert.strictEqual(drawnHandler.callCount, 0);
+            });        
+        });
+    });
+});
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.charts/chartInteraction.tests.js]: empty line is missed

File: testing/tests/DevExpress.viz.charts/chartInteraction.tests.js [hunk 4]
```diff
@@ -730,3 +730,35 @@ QUnit.test('two stub axis', function(assert) {
     assert.equal(verticalAxes[1].getOptions().grid.visible, false, 'second axis grid isn\'t visible');
     assert.equal(verticalAxes[1].getOptions().minorGrid.visible, false, 'second axis grid isn\'t visible');
 });
+
+QUnit.module('Resizing. T1156890', {
```
Comment (ksercs) [inline testing/tests/DevExpress.viz.charts/chartInteraction.tests.js]:
```suggestion
QUnit.module('Resizing (T1156890), {
```

File: js/viz/core/base_widget.js (hunk 1 above)
Comment (alexanderPolosatov) [reply js/viz/core/base_widget.js]: I agree

**Review by Krijovnick — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**